integ_test.go

```diff
@@ -149,15 +149,18 @@ WAIT:
 	goto CHECK
 }
 
-func WaitFuture(f Future, t *testing.T) error {
-	timer := time.AfterFunc(200*time.Millisecond, func() {
-		panic(fmt.Errorf("timeout waiting for future %v", f))
-	})
-	defer timer.Stop()
-	return f.Error()
+func WaitFuture(f Future, t testing.TB) error {
+	if helper, ok := t.(testingHelper); ok {
+		helper.Helper()
+	}
+	msg := "timeout waiting for future %v"
+	return waitForError(t, f.Error, 2*time.Second, msg, f)
```

Comment:
A bump from 200ms to 2s might seem high, and actually the most frequent failure that I've seen so far occurs when TestRaft_Integ runs WaitFuture(env1.raft.Snapshot(), t), as apparently Travis might be slow at syncing the snapshot file descriptor to disk. See for example [this](https://travis-ci.org/hashicorp/raft/jobs/340602304) build.

If it's preferred, we could keep the default timeout in WaitFuture lower, even the current 200ms, and add a more generic WaitFutureWIthTimeout that lets TestRaft_Integ specify a custom timeout in some call sites. WaitFuture would use it internally, and we could set the timeout to 2s only for the env1.raft.Snapshot() case (and possibly any other that might involve disk I/O).